### PR TITLE
chore: DLT-2118 fix linter issues

### PR DESCRIPTION
**Pull request by braddialpad**
# chore: DLT-2118 fix linter issues

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

- [x] Other (chore)

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-2118

## :book: Description

- Fixed any outstanding linter errors
- Updated to the latest version of stylelint-less, this seems to be able to lint everything in our utilities folder just fine so I removed lesshint, it is no longer necessary.
- Added `eslint-plugin-eslint-plugin` to the workspace package, this is used to lint eslint-plugins themselves, they were failing. `eslint-plugin-dialtone` was failing due to it not being installed.

## :bulb: Context

There were lots of linter errors happening after moving over to a monorepo, and they were no longer running on commit so more linter errors were likely added over time.

I'll do a subsequent PR to get linters running on commit / CI again since this is already quite big.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

## :crystal_ball: Next Steps

- Make linters run on commit.
- Run linters on CI

### CODE REVIEW

File: apps/dialtone-documentation/docs/.vuepress/baseComponents/SvgLoader.vue
Comment (juliodialpad) [inline apps/dialtone-documentation/docs/.vuepress/baseComponents/SvgLoader.vue:1]: Why was this change necessary?

File: apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue [hunk 1]
```diff
@@ -172,6 +172,7 @@
 </template>
 
 <script setup>
+import SvgLoader from '../../baseComponents/SvgLoader.vue';
```
Comment (juliodialpad) [inline apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue]: svgLoader is being imported gloablly on `.vuepress/client.js` file, should be ok to not import it here or was it raising a linter warning?

File: apps/dialtone-documentation/docs/.vuepress/views/IconCatalog.vue [hunk 2]
```diff
@@ -40,6 +40,7 @@
         <label for="Dialtone--SelectCategory"> Category</label>
       </div>
       <div class="d-select">
+        <!-- eslint-disable-next-line vuejs-accessibility/form-control-has-label -->
```
Comment (juliodialpad) [inline apps/dialtone-documentation/docs/.vuepress/views/IconCatalog.vue]: Why not adding the label instead of disabling the rule?

File: packages/dialtone-css/lib/build/less/components/datepicker.less [hunk 3]
```diff
@@ -14,14 +14,15 @@
...
   padding: var(--dt-space-500);
-  box-sizing: border-box;
 
   &__hd {
     padding: 0 var(--dt-space-300);
   }
 
+  /* stylelint-disable-next-line block-no-empty */
   &__bd {
   }
```
Comment (juliodialpad) [inline packages/dialtone-css/lib/build/less/components/datepicker.less]: Why not removing this?

File: packages/dialtone-css/lib/build/less/components/empty-state.less [hunk 4]
```diff
@@ -36,8 +36,10 @@
     padding: var(--dt-space-500) 0;
   }
 
+  /* stylelint-disable-next-line block-no-empty */
   &__content--md {}
 
+  /* stylelint-disable-next-line block-no-empty */
   &__content--lg {}
```
Comment (juliodialpad) [inline packages/dialtone-css/lib/build/less/components/empty-state.less]: I'd prefer to remove empty blocks rather than disabling stylelint if possible.

File: packages/dialtone-css/lib/build/less/utilities/typography.less [hunk 5]
```diff
@@ -273,8 +273,6 @@ ul {
 .d-headline48 { .d-headline--xxl(); }
 .d-headline54 { .d-headline--xxl(); }
 
-.d-fs-unset { font-size: unset !important; }
```
Comment (juliodialpad) [inline packages/dialtone-css/lib/build/less/utilities/typography.less:276]: Why are we removing this class?

File: packages/dialtone-css/package.json [hunk 6]
```diff
@@ -51,8 +51,7 @@
   "scripts": {
     "lint": "run-s lint:code lint:library",
     "lint:code": "eslint '**/*.{js,cjs,vue}'",
-    "lint:library": "run-s lesshint stylelint",
-    "lesshint": "pnpm exec lesshint 'lib/build/less/utilities'",
+    "lint:library": "run-s stylelint",
```
Comment (juliodialpad) [inline packages/dialtone-css/package.json]: Not necessary to keep `run-s` as there's only one command left to run.

Apply the suggestion and remove `stylelint` command on line 55.

```suggestion
    "lint:library": "pnpm exec stylelint 'lib/build/less/**/*.less'",
```

File: packages/dialtone-vue2/components/emoji_picker/modules/emoji_search.vue [hunk 7]
```diff
@@ -22,13 +22,13 @@
...
           size="xs"
-          class="d-mrn4"
+          class="d-emoji-picker__search-x-button"
           circle
           kind="muted"
           @click="clearSearch"
         >
           <template #icon>
-            <dt-icon-close
+            <dt-icon-x-circle
```
Comment (juliodialpad) [inline packages/dialtone-vue2/components/emoji_picker/modules/emoji_search.vue]: Is this icon change intended?

**Review by juliodialpad — CHANGES_REQUESTED** [review]

File: apps/dialtone-documentation/docs/.vuepress/baseComponents/SvgLoader.vue
Comment (braddialpad) [reply apps/dialtone-documentation/docs/.vuepress/baseComponents/SvgLoader.vue:1]: it's invalid to have a prop named "class"

File: apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue (hunk 1 above)
Comment (braddialpad) [reply apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue]: Yeah it was a linter error... There's probably some way to tell the linter about global imports so it doesn't do this.

File: packages/dialtone-css/lib/build/less/components/datepicker.less (hunk 3 above)
Comment (braddialpad) [reply packages/dialtone-css/lib/build/less/components/datepicker.less]: It looks like it was put there to define the structure so future properties could be put there. I'm fine either way, could remove I suppose

File: packages/dialtone-css/lib/build/less/utilities/typography.less (hunk 5 above)
Comment (braddialpad) [reply packages/dialtone-css/lib/build/less/utilities/typography.less:276]: Duplicate, there is another class in this file with the exact same name

File: packages/dialtone-vue2/components/emoji_picker/modules/emoji_search.vue (hunk 7 above)
Comment (braddialpad) [reply packages/dialtone-vue2/components/emoji_picker/modules/emoji_search.vue]: I think it should be close instead of x-circle. My mistake.

File: apps/dialtone-documentation/docs/.vuepress/views/IconCatalog.vue (hunk 2 above)
Comment (braddialpad) [reply apps/dialtone-documentation/docs/.vuepress/views/IconCatalog.vue]: I don't understand why this is triggering because it has a label right above it that is correctly mapped to it's ID `Dialtone--SelectCategory`

File: apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue (hunk 1 above)
Comment (braddialpad) [reply apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue]: Fixed

File: apps/dialtone-documentation/docs/.vuepress/views/IconCatalog.vue (hunk 2 above)
Comment (braddialpad) [reply apps/dialtone-documentation/docs/.vuepress/views/IconCatalog.vue]: ah, got it. needed aria-labelledby

File: packages/dialtone-css/lib/build/less/components/datepicker.less (hunk 3 above)
Comment (braddialpad) [reply packages/dialtone-css/lib/build/less/components/datepicker.less]: removed

File: packages/dialtone-css/lib/build/less/components/empty-state.less (hunk 4 above)
Comment (braddialpad) [reply packages/dialtone-css/lib/build/less/components/empty-state.less]: removed

File: apps/dialtone-documentation/docs/.vuepress/baseComponents/SvgLoader.vue
Comment (juliodialpad) [reply apps/dialtone-documentation/docs/.vuepress/baseComponents/SvgLoader.vue:1]: Ah right, I missed that, thanks for the fix!

**Review by juliodialpad — APPROVED** [review]
Looks good, thanks!
